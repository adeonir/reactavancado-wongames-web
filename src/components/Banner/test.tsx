import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import { Banner } from '.'

const props = {
  image: 'https://source.unsplash.com/user/willianjusten/1042x580',
  title: 'Defy death',
  subtitle: 'Play the new <strong>CrashLands</strong> season',
  buttonLabel: 'Buy now',
  buttonLink: '/games/defy-death',
}

describe('<Banner />', () => {
  it('should render correctly', () => {
    const {
      container: { firstChild },
    } = renderWithTheme(<Banner {...props} />)

    expect(
      screen.getByRole('heading', { name: /defy death/i }),
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', {
        name: /Play the new CrashLands season/i,
      }),
    ).toBeInTheDocument()

    expect(screen.getByRole('img')).toBeInTheDocument()

    expect(firstChild).toMatchSnapshot()
  })

  it('should render the ribbon', () => {
    const {
      container: { firstChild },
    } = renderWithTheme(
      <Banner
        {...props}
        ribbonText="Best seller"
        ribbonSize="small"
        ribbonColor="secondary"
      />,
    )

    expect(screen.getByText(/best seller/i)).toBeInTheDocument()

    expect(screen.getByText(/best seller/i)).toHaveStyle({
      backgroundColor: '#3cd3c1',
    })

    expect(screen.getByText(/best seller/i)).toHaveStyle({
      height: '2.6rem',
      fontSize: '1.2rem',
    })

    expect(firstChild).toMatchSnapshot()
  })
})
