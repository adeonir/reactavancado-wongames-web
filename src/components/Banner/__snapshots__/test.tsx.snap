// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-color: #eaeaea;
  background-image: url(https://source.unsplash.com/user/willianjusten/1042x580);
  background-position: center center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background: rgba(3,5,23,0.7);
  padding: 2.4rem;
}

.c3 {
  color: #fafafa;
  font-size: 1.4rem;
  font-weight: 400;
  margin-bottom: 1.6rem;
}

.c4 {
  color: #fafafa;
  font-size: 1.4rem;
  font-weight: 400;
  margin-bottom: 1.6rem;
}

.c4 strong {
  color: #f231a5;
  font-weight: 600;
}

.c5 {
  border-radius: 0.4rem;
  cursor: pointer;
  position: relative;
  z-index: 10;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.6rem;
  padding: 0.8rem 2.4rem;
  height: 5rem;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  border: 0.1rem solid transparent;
  color: #fafafa;
}

.c5 svg {
  width: 1.8rem;
}

.c5:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

@media (max-width:1170px) {
  .c0 .sc-bdnylx {
    right: 0;
  }

  .c0 .sc-bdnylx::after {
    display: none;
  }
}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem rgba(3,5,23,0.2);
  }
}

@media (min-width:768px) {
  .c1 {
    border-radius: 0.4rem;
    height: 58rem;
  }
}

@media (min-width:768px) {
  .c2 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<section
  class="c0"
>
  <div
    aria-label="Defy death"
    class="c1"
    role="img"
    src="https://source.unsplash.com/user/willianjusten/1042x580"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      Defy death
    </h2>
    <h3
      class="c4"
    >
      Play the new 
      <strong>
        CrashLands
      </strong>
       season
    </h3>
    <a
      class="c5"
      href="/games/defy-death"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</section>
`

exports[`<Banner /> should render the ribbon 1`] = `
.c2 {
  color: #fafafa;
  padding: 0 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 3rem;
  background-color: #3cd3c1;
  font-size: 1.2rem;
  height: 2.6rem;
  right: -1.6rem;
}

.c2::after {
  content: '';
  display: block;
  width: 0;
  height: 0;
  position: absolute;
  right: 0;
  border: 0 solid transparent;
}

.c2::after {
  border-left-color: #30a89a;
}

.c2::after {
  bottom: -0.8rem;
  border-bottom-width: 0.8rem;
  border-left-width: 1.6rem;
}

.c0 {
  position: relative;
}

.c3 {
  width: 100%;
  height: 23rem;
  background-color: #eaeaea;
  background-image: url(https://source.unsplash.com/user/willianjusten/1042x580);
  background-position: center center;
  background-size: cover;
}

.c4 {
  width: 100%;
  background: rgba(3,5,23,0.7);
  padding: 2.4rem;
}

.c5 {
  color: #fafafa;
  font-size: 1.4rem;
  font-weight: 400;
  margin-bottom: 1.6rem;
}

.c6 {
  color: #fafafa;
  font-size: 1.4rem;
  font-weight: 400;
  margin-bottom: 1.6rem;
}

.c6 strong {
  color: #f231a5;
  font-weight: 600;
}

.c7 {
  border-radius: 0.4rem;
  cursor: pointer;
  position: relative;
  z-index: 10;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.6rem;
  padding: 0.8rem 2.4rem;
  height: 5rem;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  border: 0.1rem solid transparent;
  color: #fafafa;
}

.c7 svg {
  width: 1.8rem;
}

.c7:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

@media (max-width:1170px) {
  .c0 .c1 {
    right: 0;
  }

  .c0 .c1::after {
    display: none;
  }
}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem rgba(3,5,23,0.2);
  }
}

@media (min-width:768px) {
  .c3 {
    border-radius: 0.4rem;
    height: 58rem;
  }
}

@media (min-width:768px) {
  .c4 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c6 {
    font-size: 1.8rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1 c2"
    color="secondary"
  >
    Best seller
  </div>
  <div
    aria-label="Defy death"
    class="c3"
    role="img"
    src="https://source.unsplash.com/user/willianjusten/1042x580"
  />
  <div
    class="c4"
  >
    <h2
      class="c5"
    >
      Defy death
    </h2>
    <h3
      class="c6"
    >
      Play the new 
      <strong>
        CrashLands
      </strong>
       season
    </h3>
    <a
      class="c7"
      href="/games/defy-death"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</section>
`
